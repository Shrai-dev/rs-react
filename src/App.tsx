import React, { Component } from 'react';
import { Routes, Route } from 'react-router';
import './App.css';
import Header from './components/Header/Header';
import Main from './pages/Main/Main';
import AboutUs from './pages/AboutUs/AboutUs';
import Error from './pages/Error/Error';

class App extends Component {
  render() {
    return (
      <div className="App">
        <Header />
        <Routes>
          <Route path="/" element={<Main />}>
            <Route path="main" element={<Main />} />
          </Route>
          <Route path="about-us" element={<AboutUs />} />
          <Route path="*" element={<Error />} />
        </Routes>
      </div>
    );
  }
}

export default App;
