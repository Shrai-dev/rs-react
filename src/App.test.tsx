import { describe, it } from 'vitest';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import App from './App';
import React from 'react';

describe('App', () => {
  it('should render App and match snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <App />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('Render Not Found page'),
    () => {
      render(
        <MemoryRouter initialEntries={['/this-route-does-not-exist']}>
          <App />
        </MemoryRouter>
      );
      expect(
        screen.getByRole('heading', {
          level: 2,
        })
      ).toHaveTextContent('Page not found');
    };
});
