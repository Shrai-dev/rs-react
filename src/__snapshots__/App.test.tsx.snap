// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > should render App and match snapshot 1`] = `
<DocumentFragment>
  <div
    class="app__container"
  >
    <header
      class="header"
    >
      <nav
        class="nav__bar"
      >
        <ul
          class="nav__list"
        >
          <li>
            <a
              class="backward__link"
              href="/about-us"
            >
              About Us
            </a>
          </li>
          <li>
            <a
              class="backward__link"
              href="/main"
            >
              Main
            </a>
          </li>
        </ul>
      </nav>
    </header>
    <main
      class="main"
    >
      <div
        class="search__container"
      >
        <form>
          <input
            class="search__input"
            id=""
            name="searchValue"
            type="search"
            value=""
          />
          <button
            class="search__btn"
            type="submit"
          >
            Search
          </button>
        </form>
      </div>
      <div
        class="cards__wrapper"
      >
        <div
          class="card__item"
        >
          <div
            class="card__item-img"
          >
            <img
              alt=""
              class="card__img"
              src="https://i.dummyjson.com/data/products/1/1.jpg"
            />
          </div>
          <h3
            class="card__item-title"
          >
            iPhone 9
          </h3>
          <h4
            class="card__item-subtitle"
          >
            Brand: Apple
          </h4>
          <p
            class="card__item-desc"
          >
            An apple mobile which is nothing like apple
          </p>
          <div
            class="card__item-details"
          >
            <p
              class="card__item-info"
            >
              Rating: 4.69
            </p>
            <p
              class="card__item-info"
            >
              Price: $549
            </p>
            <p
              class="card__item-info"
            >
              Discount: 0%
            </p>
            <p
              class="card__item-info"
            >
              Stock: 94
            </p>
          </div>
          <button
            class="card__item-btn"
          >
            Show more
          </button>
        </div>
        <div
          class="card__item"
        >
          <div
            class="card__item-img"
          >
            <img
              alt=""
              class="card__img"
              src="https://i.dummyjson.com/data/products/2/1.jpg"
            />
          </div>
          <h3
            class="card__item-title"
          >
            iPhone X
          </h3>
          <h4
            class="card__item-subtitle"
          >
            Brand: Apple
          </h4>
          <p
            class="card__item-desc"
          >
            SIM-Free, Model A19211 6.5-inch Super Retina HD display with OLED technology A12 Bionic chip with ...
          </p>
          <div
            class="card__item-details"
          >
            <p
              class="card__item-info"
            >
              Rating: 4.44
            </p>
            <p
              class="card__item-info"
            >
              Price: $899
            </p>
            <p
              class="card__item-info"
            >
              Discount: 0%
            </p>
            <p
              class="card__item-info"
            >
              Stock: 34
            </p>
          </div>
          <button
            class="card__item-btn"
          >
            Show more
          </button>
        </div>
        <div
          class="card__item"
        >
          <div
            class="card__item-img"
          >
            <img
              alt=""
              class="card__img"
              src="https://i.dummyjson.com/data/products/3/1.jpg"
            />
          </div>
          <h3
            class="card__item-title"
          >
            Samsung Universe 9
          </h3>
          <h4
            class="card__item-subtitle"
          >
            Brand: Samsung
          </h4>
          <p
            class="card__item-desc"
          >
            Samsung's new variant which goes beyond Galaxy to the Universe
          </p>
          <div
            class="card__item-details"
          >
            <p
              class="card__item-info"
            >
              Rating: 4.09
            </p>
            <p
              class="card__item-info"
            >
              Price: $1249
            </p>
            <p
              class="card__item-info"
            >
              Discount: 0%
            </p>
            <p
              class="card__item-info"
            >
              Stock: 36
            </p>
          </div>
          <button
            class="card__item-btn"
          >
            Show more
          </button>
        </div>
        <div
          class="card__item"
        >
          <div
            class="card__item-img"
          >
            <img
              alt=""
              class="card__img"
              src="https://i.dummyjson.com/data/products/4/1.jpg"
            />
          </div>
          <h3
            class="card__item-title"
          >
            OPPOF19
          </h3>
          <h4
            class="card__item-subtitle"
          >
            Brand: OPPO
          </h4>
          <p
            class="card__item-desc"
          >
            OPPO F19 is officially announced on April 2021.
          </p>
          <div
            class="card__item-details"
          >
            <p
              class="card__item-info"
            >
              Rating: 4.3
            </p>
            <p
              class="card__item-info"
            >
              Price: $280
            </p>
            <p
              class="card__item-info"
            >
              Discount: 0%
            </p>
            <p
              class="card__item-info"
            >
              Stock: 123
            </p>
          </div>
          <button
            class="card__item-btn"
          >
            Show more
          </button>
        </div>
        <div
          class="card__item"
        >
          <div
            class="card__item-img"
          >
            <img
              alt=""
              class="card__img"
              src="https://i.dummyjson.com/data/products/5/1.jpg"
            />
          </div>
          <h3
            class="card__item-title"
          >
            Huawei P30
          </h3>
          <h4
            class="card__item-subtitle"
          >
            Brand: Huawei
          </h4>
          <p
            class="card__item-desc"
          >
            Huaweiâ€™s re-badged P30 Pro New Edition was officially unveiled yesterday in Germany and now the device has made its way to the UK.
          </p>
          <div
            class="card__item-details"
          >
            <p
              class="card__item-info"
            >
              Rating: 4.09
            </p>
            <p
              class="card__item-info"
            >
              Price: $499
            </p>
            <p
              class="card__item-info"
            >
              Discount: 0%
            </p>
            <p
              class="card__item-info"
            >
              Stock: 32
            </p>
          </div>
          <button
            class="card__item-btn"
          >
            Show more
          </button>
        </div>
        <div
          class="card__item"
        >
          <div
            class="card__item-img"
          >
            <img
              alt=""
              class="card__img"
              src="https://i.dummyjson.com/data/products/6/1.png"
            />
          </div>
          <h3
            class="card__item-title"
          >
            MacBook Pro
          </h3>
          <h4
            class="card__item-subtitle"
          >
            Brand: Apple
          </h4>
          <p
            class="card__item-desc"
          >
            MacBook Pro 2021 with mini-LED display may launch between September, November
          </p>
          <div
            class="card__item-details"
          >
            <p
              class="card__item-info"
            >
              Rating: 4.57
            </p>
            <p
              class="card__item-info"
            >
              Price: $1749
            </p>
            <p
              class="card__item-info"
            >
              Discount: 0%
            </p>
            <p
              class="card__item-info"
            >
              Stock: 83
            </p>
          </div>
          <button
            class="card__item-btn"
          >
            Show more
          </button>
        </div>
        <div
          class="card__item"
        >
          <div
            class="card__item-img"
          >
            <img
              alt=""
              class="card__img"
              src="https://i.dummyjson.com/data/products/7/1.jpg"
            />
          </div>
          <h3
            class="card__item-title"
          >
            Samsung Galaxy Book
          </h3>
          <h4
            class="card__item-subtitle"
          >
            Brand: Samsung
          </h4>
          <p
            class="card__item-desc"
          >
            Samsung Galaxy Book S (2020) Laptop With Intel Lakefield Chip, 8GB of RAM Launched
          </p>
          <div
            class="card__item-details"
          >
            <p
              class="card__item-info"
            >
              Rating: 4.25
            </p>
            <p
              class="card__item-info"
            >
              Price: $1499
            </p>
            <p
              class="card__item-info"
            >
              Discount: 0%
            </p>
            <p
              class="card__item-info"
            >
              Stock: 50
            </p>
          </div>
          <button
            class="card__item-btn"
          >
            Show more
          </button>
        </div>
        <div
          class="card__item"
        >
          <div
            class="card__item-img"
          >
            <img
              alt=""
              class="card__img"
              src="https://i.dummyjson.com/data/products/8/1.jpg"
            />
          </div>
          <h3
            class="card__item-title"
          >
            Microsoft Surface Laptop 4
          </h3>
          <h4
            class="card__item-subtitle"
          >
            Brand: Microsoft Surface
          </h4>
          <p
            class="card__item-desc"
          >
            Style and speed. Stand out on HD video calls backed by Studio Mics. Capture ideas on the vibrant touchscreen.
          </p>
          <div
            class="card__item-details"
          >
            <p
              class="card__item-info"
            >
              Rating: 4.43
            </p>
            <p
              class="card__item-info"
            >
              Price: $1499
            </p>
            <p
              class="card__item-info"
            >
              Discount: 0%
            </p>
            <p
              class="card__item-info"
            >
              Stock: 68
            </p>
          </div>
          <button
            class="card__item-btn"
          >
            Show more
          </button>
        </div>
        <div
          class="card__item"
        >
          <div
            class="card__item-img"
          >
            <img
              alt=""
              class="card__img"
              src="https://i.dummyjson.com/data/products/9/1.jpg"
            />
          </div>
          <h3
            class="card__item-title"
          >
            Infinix INBOOK
          </h3>
          <h4
            class="card__item-subtitle"
          >
            Brand: Infinix
          </h4>
          <p
            class="card__item-desc"
          >
            Infinix Inbook X1 Ci3 10th 8GB 256GB 14 Win10 Grey â€“ 1 Year Warranty
          </p>
          <div
            class="card__item-details"
          >
            <p
              class="card__item-info"
            >
              Rating: 4.54
            </p>
            <p
              class="card__item-info"
            >
              Price: $1099
            </p>
            <p
              class="card__item-info"
            >
              Discount: 0%
            </p>
            <p
              class="card__item-info"
            >
              Stock: 96
            </p>
          </div>
          <button
            class="card__item-btn"
          >
            Show more
          </button>
        </div>
        <div
          class="card__item"
        >
          <div
            class="card__item-img"
          >
            <img
              alt=""
              class="card__img"
              src="https://i.dummyjson.com/data/products/10/1.jpg"
            />
          </div>
          <h3
            class="card__item-title"
          >
            HP Pavilion 15-DK1056WM
          </h3>
          <h4
            class="card__item-subtitle"
          >
            Brand: HP Pavilion
          </h4>
          <p
            class="card__item-desc"
          >
            HP Pavilion 15-DK1056WM Gaming Laptop 10th Gen Core i5, 8GB, 256GB SSD, GTX 1650 4GB, Windows 10
          </p>
          <div
            class="card__item-details"
          >
            <p
              class="card__item-info"
            >
              Rating: 4.43
            </p>
            <p
              class="card__item-info"
            >
              Price: $1099
            </p>
            <p
              class="card__item-info"
            >
              Discount: 0%
            </p>
            <p
              class="card__item-info"
            >
              Stock: 89
            </p>
          </div>
          <button
            class="card__item-btn"
          >
            Show more
          </button>
        </div>
        <div
          class="card__item"
        >
          <div
            class="card__item-img"
          >
            <img
              alt=""
              class="card__img"
              src="https://i.dummyjson.com/data/products/11/1.jpg"
            />
          </div>
          <h3
            class="card__item-title"
          >
            perfume Oil
          </h3>
          <h4
            class="card__item-subtitle"
          >
            Brand: Impression of Acqua Di Gio
          </h4>
          <p
            class="card__item-desc"
          >
            Mega Discount, Impression of Acqua Di Gio by GiorgioArmani concentrated attar perfume Oil
          </p>
          <div
            class="card__item-details"
          >
            <p
              class="card__item-info"
            >
              Rating: 4.26
            </p>
            <p
              class="card__item-info"
            >
              Price: $13
            </p>
            <p
              class="card__item-info"
            >
              Discount: 0%
            </p>
            <p
              class="card__item-info"
            >
              Stock: 65
            </p>
          </div>
          <button
            class="card__item-btn"
          >
            Show more
          </button>
        </div>
        <div
          class="card__item"
        >
          <div
            class="card__item-img"
          >
            <img
              alt=""
              class="card__img"
              src="https://i.dummyjson.com/data/products/12/1.jpg"
            />
          </div>
          <h3
            class="card__item-title"
          >
            Brown Perfume
          </h3>
          <h4
            class="card__item-subtitle"
          >
            Brand: Royal_Mirage
          </h4>
          <p
            class="card__item-desc"
          >
            Royal_Mirage Sport Brown Perfume for Men & Women - 120ml
          </p>
          <div
            class="card__item-details"
          >
            <p
              class="card__item-info"
            >
              Rating: 4
            </p>
            <p
              class="card__item-info"
            >
              Price: $40
            </p>
            <p
              class="card__item-info"
            >
              Discount: 0%
            </p>
            <p
              class="card__item-info"
            >
              Stock: 52
            </p>
          </div>
          <button
            class="card__item-btn"
          >
            Show more
          </button>
        </div>
        <div
          class="card__item"
        >
          <div
            class="card__item-img"
          >
            <img
              alt=""
              class="card__img"
              src="https://i.dummyjson.com/data/products/13/1.jpg"
            />
          </div>
          <h3
            class="card__item-title"
          >
            Fog Scent Xpressio Perfume
          </h3>
          <h4
            class="card__item-subtitle"
          >
            Brand: Fog Scent Xpressio
          </h4>
          <p
            class="card__item-desc"
          >
            Product details of Best Fog Scent Xpressio Perfume 100ml For Men cool long lasting perfumes for Men
          </p>
          <div
            class="card__item-details"
          >
            <p
              class="card__item-info"
            >
              Rating: 4.59
            </p>
            <p
              class="card__item-info"
            >
              Price: $13
            </p>
            <p
              class="card__item-info"
            >
              Discount: 0%
            </p>
            <p
              class="card__item-info"
            >
              Stock: 61
            </p>
          </div>
          <button
            class="card__item-btn"
          >
            Show more
          </button>
        </div>
        <div
          class="card__item"
        >
          <div
            class="card__item-img"
          >
            <img
              alt=""
              class="card__img"
              src="https://i.dummyjson.com/data/products/14/1.jpg"
            />
          </div>
          <h3
            class="card__item-title"
          >
            Non-Alcoholic Concentrated Perfume Oil
          </h3>
          <h4
            class="card__item-subtitle"
          >
            Brand: Al Munakh
          </h4>
          <p
            class="card__item-desc"
          >
            Original Al MunakhÂ® by Mahal Al Musk | Our Impression of Climate | 6ml Non-Alcoholic Concentrated Perfume Oil
          </p>
          <div
            class="card__item-details"
          >
            <p
              class="card__item-info"
            >
              Rating: 4.21
            </p>
            <p
              class="card__item-info"
            >
              Price: $120
            </p>
            <p
              class="card__item-info"
            >
              Discount: 0%
            </p>
            <p
              class="card__item-info"
            >
              Stock: 114
            </p>
          </div>
          <button
            class="card__item-btn"
          >
            Show more
          </button>
        </div>
        <div
          class="card__item"
        >
          <div
            class="card__item-img"
          >
            <img
              alt=""
              class="card__img"
              src="https://i.dummyjson.com/data/products/15/1.jpg"
            />
          </div>
          <h3
            class="card__item-title"
          >
            Eau De Perfume Spray
          </h3>
          <h4
            class="card__item-subtitle"
          >
            Brand: Lord - Al-Rehab
          </h4>
          <p
            class="card__item-desc"
          >
            Genuine  Al-Rehab spray perfume from UAE/Saudi Arabia/Yemen High Quality
          </p>
          <div
            class="card__item-details"
          >
            <p
              class="card__item-info"
            >
              Rating: 4.7
            </p>
            <p
              class="card__item-info"
            >
              Price: $30
            </p>
            <p
              class="card__item-info"
            >
              Discount: 0%
            </p>
            <p
              class="card__item-info"
            >
              Stock: 105
            </p>
          </div>
          <button
            class="card__item-btn"
          >
            Show more
          </button>
        </div>
        <div
          class="card__item"
        >
          <div
            class="card__item-img"
          >
            <img
              alt=""
              class="card__img"
              src="https://i.dummyjson.com/data/products/16/1.png"
            />
          </div>
          <h3
            class="card__item-title"
          >
            Hyaluronic Acid Serum
          </h3>
          <h4
            class="card__item-subtitle"
          >
            Brand: L'Oreal Paris
          </h4>
          <p
            class="card__item-desc"
          >
            L'OrÃƒÂ©al Paris introduces Hyaluron Expert Replumping Serum formulated with 1.5% Hyaluronic Acid
          </p>
          <div
            class="card__item-details"
          >
            <p
              class="card__item-info"
            >
              Rating: 4.83
            </p>
            <p
              class="card__item-info"
            >
              Price: $19
            </p>
            <p
              class="card__item-info"
            >
              Discount: 0%
            </p>
            <p
              class="card__item-info"
            >
              Stock: 110
            </p>
          </div>
          <button
            class="card__item-btn"
          >
            Show more
          </button>
        </div>
        <div
          class="card__item"
        >
          <div
            class="card__item-img"
          >
            <img
              alt=""
              class="card__img"
              src="https://i.dummyjson.com/data/products/17/1.jpg"
            />
          </div>
          <h3
            class="card__item-title"
          >
            Tree Oil 30ml
          </h3>
          <h4
            class="card__item-subtitle"
          >
            Brand: Hemani Tea
          </h4>
          <p
            class="card__item-desc"
          >
            Tea tree oil contains a number of compounds, including terpinen-4-ol, that have been shown to kill certain bacteria,
          </p>
          <div
            class="card__item-details"
          >
            <p
              class="card__item-info"
            >
              Rating: 4.52
            </p>
            <p
              class="card__item-info"
            >
              Price: $12
            </p>
            <p
              class="card__item-info"
            >
              Discount: 0%
            </p>
            <p
              class="card__item-info"
            >
              Stock: 78
            </p>
          </div>
          <button
            class="card__item-btn"
          >
            Show more
          </button>
        </div>
        <div
          class="card__item"
        >
          <div
            class="card__item-img"
          >
            <img
              alt=""
              class="card__img"
              src="https://i.dummyjson.com/data/products/18/1.jpg"
            />
          </div>
          <h3
            class="card__item-title"
          >
            Oil Free Moisturizer 100ml
          </h3>
          <h4
            class="card__item-subtitle"
          >
            Brand: Dermive
          </h4>
          <p
            class="card__item-desc"
          >
            Dermive Oil Free Moisturizer with SPF 20 is specifically formulated with ceramides, hyaluronic acid & sunscreen.
          </p>
          <div
            class="card__item-details"
          >
            <p
              class="card__item-info"
            >
              Rating: 4.56
            </p>
            <p
              class="card__item-info"
            >
              Price: $40
            </p>
            <p
              class="card__item-info"
            >
              Discount: 0%
            </p>
            <p
              class="card__item-info"
            >
              Stock: 88
            </p>
          </div>
          <button
            class="card__item-btn"
          >
            Show more
          </button>
        </div>
        <div
          class="card__item"
        >
          <div
            class="card__item-img"
          >
            <img
              alt=""
              class="card__img"
              src="https://i.dummyjson.com/data/products/19/1.jpg"
            />
          </div>
          <h3
            class="card__item-title"
          >
            Skin Beauty Serum.
          </h3>
          <h4
            class="card__item-subtitle"
          >
            Brand: ROREC White Rice
          </h4>
          <p
            class="card__item-desc"
          >
            Product name: rorec collagen hyaluronic acid white face serum riceNet weight: 15 m
          </p>
          <div
            class="card__item-details"
          >
            <p
              class="card__item-info"
            >
              Rating: 4.42
            </p>
            <p
              class="card__item-info"
            >
              Price: $46
            </p>
            <p
              class="card__item-info"
            >
              Discount: 0%
            </p>
            <p
              class="card__item-info"
            >
              Stock: 54
            </p>
          </div>
          <button
            class="card__item-btn"
          >
            Show more
          </button>
        </div>
        <div
          class="card__item"
        >
          <div
            class="card__item-img"
          >
            <img
              alt=""
              class="card__img"
              src="https://i.dummyjson.com/data/products/20/1.jpg"
            />
          </div>
          <h3
            class="card__item-title"
          >
            Freckle Treatment Cream- 15gm
          </h3>
          <h4
            class="card__item-subtitle"
          >
            Brand: Fair & Clear
          </h4>
          <p
            class="card__item-desc"
          >
            Fair & Clear is Pakistan's only pure Freckle cream which helpsfade Freckles, Darkspots and pigments. Mercury level is 0%, so there are no side effects.
          </p>
          <div
            class="card__item-details"
          >
            <p
              class="card__item-info"
            >
              Rating: 4.06
            </p>
            <p
              class="card__item-info"
            >
              Price: $70
            </p>
            <p
              class="card__item-info"
            >
              Discount: 0%
            </p>
            <p
              class="card__item-info"
            >
              Stock: 140
            </p>
          </div>
          <button
            class="card__item-btn"
          >
            Show more
          </button>
        </div>
      </div>
    </main>
  </div>
</DocumentFragment>
`;
